app/component/itinerary/ItineraryPage.js

```diff
@@ -1130,10 +1154,9 @@ export default function ItineraryPage(props, context) {
             />
           )}
           <NaviContainer
-            itinerary={combinedEdges[selectedIndex]?.node}
+            itinerary={naviModeItinerary}
```

Comment:
I noticed that NaviContainer actually only needs the legs array. Time to change the prop to legs?

Reply:
I recall making the same observation during development as well, but I cannot remember why I did not include it as a tentative change. I'll refactor to pass legs only if possible.